### PR TITLE
fix: cleaning up processing code

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- Keep the code clean.

## Changes

- use protocol in interfaces to make mypy happy.
- Add missing typing and functions to interfaces
- remove unused code from processing downloader.
- rename processing.logger to processing.logging_util.py because logger is a name often used to refer to the module level logger.`

## Testing steps

- unit tests

### CODE REVIEW

File: backend/layers/persistence/persistence_interface.py [hunk 1]
```diff
@@ -146,16 +146,6 @@ def finalize_collection_version(
         2. set_published_at
         """
 
-    def update_collection_version_mapping(self, collection_id: CollectionId, version_id: CollectionVersionId) -> None:
```
Comment (Bento007) [inline backend/layers/persistence/persistence_interface.py:149]: not used outside of this interface

File: backend/layers/persistence/persistence_interface.py [hunk 2]
```diff
@@ -146,16 +146,6 @@ def finalize_collection_version(
...
-        """
-        Updates the mapping between the canonical collection `collection_id` and its `version_id`
-        """
-
-    def set_collection_version_published_at(self, version_id: CollectionVersionId, published_at: datetime) -> None:
```
Comment (Bento007) [inline backend/layers/persistence/persistence_interface.py:154]: not used outside of this interface

File: backend/layers/processing/downloader.py [hunk 3]
```diff
@@ -12,59 +11,6 @@
 logger = logging.getLogger("processing")
 
 
-class ProgressTracker:
```
Comment (Bento007) [inline backend/layers/processing/downloader.py:15]: not used

File: backend/layers/processing/downloader.py [hunk 4]
```diff
@@ -12,59 +11,6 @@
...
-        with self.progress_lock:
-            self._progress += progress
-
-    def cancel(self):
-        self.tombstoned = True
-        self.stop_downloader.set()
-        self.stop_updater.set()
-
-
-class NoOpProgressTracker:
```
Comment (Bento007) [inline backend/layers/processing/downloader.py:39]: not used

File: backend/layers/processing/process_logic.py [hunk 5]
```diff
@@ -12,12 +12,12 @@
 )
 from backend.layers.processing.downloader import Downloader
 from backend.layers.processing.exceptions import ConversionFailed
-from backend.layers.processing.logger import logit
+from backend.layers.processing.logging_utils import logit
 from backend.layers.thirdparty.s3_provider import S3ProviderInterface
 from backend.layers.thirdparty.uri_provider import UriProviderInterface
 
 
-class ProcessingLogic:  # TODO: ProcessingLogicBase
```
Comment (Bento007) [inline backend/layers/processing/process_logic.py:20]: Ingestion is more descriptive than Processing

File: backend/layers/thirdparty/crossref_provider.py [hunk 6]
```diff
@@ -159,3 +160,23 @@ def fetch_preprint_published_doi(self, doi):
                     return published_doi[0]["id"]
             except Exception:
                 pass
+
+
+class MockCrossrefProvider(CrossrefProviderInterface):
```
Comment (Bento007) [inline backend/layers/thirdparty/crossref_provider.py:165]: Needed to fix tests that generate a collections as a result of making the interfaces protocol

File: backend/layers/thirdparty/s3_provider_mock.py [hunk 7]
```diff
@@ -55,3 +55,9 @@ def uri_exists(self, uri: str):
 
     def is_empty(self):
         return len(self.mock_s3_fs) == 0
+
+    def generate_presigned_url(self, path: str, expiration: int = 604800) -> str:
```
Comment (Bento007) [inline backend/layers/thirdparty/s3_provider_mock.py:59]: These function definition are need for mypy to recognize this as using the S3ProviderInterface

File: tests/unit/backend/layers/api/test_curation_api.py [hunk 8]
```diff
@@ -559,7 +559,8 @@ def confirm_timestamp_fields_present_then_remove(self, received_body: dict):
         # Confirm fields are present on Collection version body but ignore equality comparison for timestamps
         self.assertIn("created_at", received_body)
         self.assertIn("published_at", received_body)
-        [self.assertIn("published_at", d) for d in received_body["dataset_versions"]]
+        for d in received_body["dataset_versions"]:
```
Comment (Bento007) [inline tests/unit/backend/layers/api/test_curation_api.py:562]: mypy fix

**Comment by danieljhegeman** [comment]
@Bento007 I didn't realize we had an auto-closer. I guess another opportunity to check on the rdev cleanup 😄 

I will get to this PR eventually I promise

**Pull request closed without merging**